packages/x-tree-view/src/TreeItem2/TreeItem2.tsx

```diff
@@ -155,6 +167,48 @@ export const TreeItem2GroupTransition = styled(Collapse, {
   ],
 });
 
+function UnstyledLabelInput({ onChange, value, ...props }: UseTreeItem2LabelInputSlotOwnProps) {
+  const [labelInputValue, setLabelInputValue] = React.useState(value);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    onChange?.(event);
+    setLabelInputValue(event.target.value);
+  };
+
+  const resetLabel = React.useCallback(() => setLabelInputValue(value), [value]);
+
+  React.useEffect(() => {
```

Comment:
This would behave the same I think:

```suggestion
  React.useEffect(() => {
    return () => setLabelInputValue(value)
  }, [value]);
```

____


Don't you think that it's something that belongs inside our plugins rather than in the component?

Reply:
> This would behave the same I think

Yeah you are right

> Don't you think that it's something that belongs inside our plugins rather than in the component?

I'm trying to think of a way that this could work inside the plugins 🤔 Would you move this small state to the plugin too?

Reply:
I'm not sure
I like the idea that users can pass a custom UI without caring about this logic.

As discussed last week (I think), I would move all the label input props to a `useTreeViewLabel.itemPlugin.ts` using the props enhancer.
And I would move the state their as well.
This would also allow not to bundle all of this logic on the `SimpleTreeView`